src/transformers/models/character_bert/modeling_character_bert.py

```diff
@@ -0,0 +1,1938 @@
+# coding=utf-8
+# Copyright Hicham EL BOUKKOURI, Olivier FERRET, Thomas LAVERGNE, Hiroshi NOJI,
+# Pierre ZWEIGENBAUM, Junichi TSUJII, The HuggingFace Inc. and AllenNLP teams.
+# All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch CharacterBERT model: this is a variant of BERT that uses the
+CharacterCNN module from ELMo instead of a WordPiece embedding matrix.
+See: “CharacterBERT: Reconciling ELMo and BERT for Word-Level Open-Vocabulary
+Representations From Characters“ https://www.aclweb.org/anthology/2020.coling-main.609/ """
+import math
+import warnings
+from dataclasses import dataclass
+from typing import Callable, Optional, Tuple
+
+import torch
+import torch.utils.checkpoint
+from torch import nn
+from torch.nn import CrossEntropyLoss, MSELoss
+
+from ...activations import ACT2FN
+from ...file_utils import (
+    ModelOutput,
+    add_code_sample_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    replace_return_docstrings,
+)
+from ...modeling_outputs import (
+    BaseModelOutputWithPastAndCrossAttentions,
+    BaseModelOutputWithPoolingAndCrossAttentions,
+    CausalLMOutputWithCrossAttentions,
+    MaskedLMOutput,
+    MultipleChoiceModelOutput,
+    NextSentencePredictorOutput,
+    QuestionAnsweringModelOutput,
+    SequenceClassifierOutput,
+    TokenClassifierOutput,
+)
+from ...modeling_utils import (
+    PreTrainedModel,
+    apply_chunking_to_forward,
+    find_pruneable_heads_and_indices,
+    prune_linear_layer,
+)
+from ...utils import logging
+from .configuration_character_bert import CharacterBertConfig
+from .tokenization_character_bert import CharacterMapper
+
+
+logger = logging.get_logger(__name__)
+
+_CONFIG_FOR_DOC = "CharacterBertConfig"
+_TOKENIZER_FOR_DOC = "CharacterBertTokenizer"
+
+CHARACTER_BERT_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "helboukkouri/character-bert",
+    "helboukkouri/character-bert-medical",
+    # See all CharacterBERT models at https://huggingface.co/models?filter=character_bert
+]
+
+
+class Highway(torch.nn.Module):
+    """
+    NOTE: taken from https://github.com/allenai/allennlp/blob/main/allennlp/modules/highway.py
```

Comment:
This should be set as a comment above: attributions are usually done as comment in the source code in the library, not documentation.